src/script/interpreter.h

```diff
@@ -160,10 +162,11 @@ class BaseSignatureChecker
     virtual ~BaseSignatureChecker() {}
 };
 
-class TransactionSignatureChecker : public BaseSignatureChecker
+template<class T>
+class GenericTransactionSignatureChecker : public BaseSignatureChecker
```

Comment:
nit, drop `Generic` prefix?

Reply:
I've added it so that the names `TransactionSignatureChecker` and `MutableTransactionSignatureChecker` are still valid as used before

Reply:
Right.